src/util/javaServerMode.ts

```diff
@@ -0,0 +1,50 @@
+import { window, commands } from "vscode";
+import { JavaExtensionAPI } from "../extension";
+
+export const JAVA_EXTENSION_ID = "redhat.java";
+
+export enum ServerMode {
+    STANDARD = "Standard",
+    LIGHTWEIGHT = "LightWeight",
+    HYBRID = "Hybrid",
+}
+
+/**
+ * Waits for the java language server to launch in standard mode
+ * Before activating Tools for MicroProfile.
+ * If java ls was started in lightweight mode, It will prompt user to switch
+ */
+export async function waitForStandardMode(api: JavaExtensionAPI): Promise<void> {
```

Comment:
license header needed. If part of this code is re-used from another open-source project we should credit the original license as well